src/NeoSharp.Core/SmartContract/Invocation/InvocationProcess.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using NeoSharp.Core.Blockchain;
+using NeoSharp.Core.Models;
+using NeoSharp.Core.SmartContract.ContractParameters;
+using NeoSharp.Core.Types;
+using NeoSharp.VM;
+
+namespace NeoSharp.Core.SmartContract
+{
+    public class InvocationProcess
+    {
+        private IExecutionEngine _executionEngine;
+
+        public InvocationProcess(IExecutionEngine executionEngine)
+        {
+            _executionEngine = executionEngine;
+        }
+
+        /// <summary>
+        /// Invokes the specified Contract using the provided parameters.
+        /// </summary>
+        /// <returns>The invoke.</returns>
+        /// <param name="scriptHash">Script hash.</param>
+        /// <param name="contractParameters">Contract parameters.</param>
+        public InvocationResult Invoke(UInt160 scriptHash, ContractParameter[] contractParameters)
+        {
+            var invocationScript = BuildInvocationScript(scriptHash, contractParameters);
+            var invocationResult = RunInvocationOnVM(invocationScript);
+            return invocationResult;
+        }
+
+        /// <summary>
+        /// Concatenates the scripts and runs it in a single invocation.
+        /// </summary>
+        /// <returns>The multiple.</returns>
+        /// <param name="contractParameterMap">Contract parameter map.</param>
+        public InvocationResult InvokeMultiple(Dictionary<UInt160, ContractParameter[]> contractParameterMap)

```

Comment:
This assumes that the each invocation is on a different contract but that is not necessarily the case. 

The call to determine a balance of an address requires 2 calls to the same contract: one `decimals` and one `balanceOf`. I dont think a dictionary will work in this case.